frontend/src/utils/blockUtils.js

```diff
@@ -33,6 +33,13 @@ export function generateId(block) {
   return id;
 }
 
+export function generatePipelineId(pipeline) {
```

Comment:
We also generate IDs for the pipeline here: https://github.com/zetane/ZetaForge/blob/aa67b80304f5165fc2fa6b5dd87ecf4ab0d872b5/frontend/src/atoms/pipelineAtom.js#L16-L19. It would be nice to plug that function in there as well to make sure all the pipelines have their IDs generated in the same way.

Reply:
solved it like this way:
```
const id = generatePipelineId(pipeline || { name: 'pipeline' });
const buffer = `${cachePath}${id}`;
```